File: JobDescription/src/src/validator/JobValidator.java
package validator;

import domain.Job;
import exception.JobException;

import java.util.Arrays;

/**
 * Created by claudiu on 12.10.2016.
 */
public class JobValidator {
    private String[] types = new String[]{"full time", "part time"};
    public void validate(Job job) throws JobException {
        if (job.getName().isEmpty()) {
            throw new JobException("Name can't be empty!");
        }

        if (job.getType().isEmpty()) {
            throw new JobException("Type can't be empty!");
        }

        int a = Arrays.binarySearch(this.types, job.getType().toString());
        if (a > 1) {
            throw new JobException("Type can't be " + job.getType() + " !");
        }

    }
}

File: JobDescriptionCsharp/JobDescriptionCsharp/validator/JobValidator.cs
using JobDescriptionCsharp.domain;
using JobDescriptionCsharp.exception;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JobDescriptionCsharp.validator
{
    public class JobValidator : IValidator<Job>
    {
        public void validate(Job job)
        {
            if (job.name == "")
                throw new JobException("The name of job can't be empty!");

            if (job.type == "")
                throw new JobException("The type of job can't be empty!");

            if (!(job.type.Equals("full-time") || job.type.Equals("part-time")))
                throw new JobException("The type must be full-time or part-time!");
        }
    }
}
